tests/test_hf_api.py

```diff
@@ -450,6 +451,28 @@ def test_token_workflow(self):
         self.assertEqual(HfFolder.get_token(), None)
 
 
+class HfApiPrivateTest(HfApiCommonTestWithLogin):
```

Comment:
@osanseviero here is a new unit test for probing the model info with private repos. 

I'm not sure if this is exactly what you had in mind, but if it is OK I'll remove the existing test from `snapshot_download`